src/Model/Bolt/Api/CreateOrder.php

```diff
@@ -150,6 +151,7 @@ public function execute(
             $immutableQuote = $this->loadQuoteData($quoteId);
 
             $this->preProcessWebhook($immutableQuote->getStoreId());
+            $immutableQuote->getStore()->setCurrentCurrencyCode($immutableQuote->getQuoteCurrencyCode());
```

Comment:
interesting. Normally what immutable means is unchangeable. Here you are changing it ^_^ This is a cause for concern. Is there any other way around this or it's the only way?